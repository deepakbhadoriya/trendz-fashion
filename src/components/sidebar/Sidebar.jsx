import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { FormattedMessage, useIntl } from 'react-intl';

import MensClothingIcon from '../../assets/icons/sidebar/mensClothingIcon';
import WomenClothingIcon from '../../assets/icons/sidebar/WomenClothingIcon';
import JewelleryIcon from '../../assets/icons/sidebar/jewelleryIcon';
import styles from './styles.module.scss';
import ElectronicsIcon from '../../assets/icons/sidebar/electronicsIcon';
import LogoutIcon from '../../assets/icons/logout/logoutIcon';
import HelpIcon from '../../assets/icons/help/helpIcon';
import ArrowLeftIcon from '../../assets/icons/arrows/arrowLeft';
import { $id } from '../../utils/domUtils';
import LogInIcon from '../../assets/icons/login/loginIcon';

const Sidebar = ({ openLogin, isLoggedIn, currentUser, handleLogout }) => {
  const intl = useIntl();
  const locale = useSelector((state) => state.language.locale);

  const [sidebarVisible, toggleSidebar] = useState(() => window.innerWidth >= 900);

  useEffect(() => {
    const rootDiv = $id('root');
    if (rootDiv) {
      rootDiv.classList.remove('sidebarOpen');
      rootDiv.classList.remove('sidebarMinimized');
      rootDiv.classList.add(sidebarVisible ? 'sidebarOpen' : 'sidebarMinimized');
    }
  }, [sidebarVisible]);

  const updateSideBarVisibility = (event) => {
    toggleSidebar(!sidebarVisible);
  };

  return (
    <div
      className={`${styles.sidebar} ${sidebarVisible ? styles.sidebarFull : styles.sidebarMinimized
        }`}
    >
      <button
        aria-label="Sidebar toggle"
        className={`${styles.sidebarToggler} ${sidebarVisible ? '' : styles.sidebarHidden}`}
        onClick={updateSideBarVisibility}
      >
        <ArrowLeftIcon />
      </button>
      <div className={styles.navLinks}>
        <Link
          to="/shop/men"
          data-test-id="menCategory"
          className={styles.navLink}
          data-target="mens"
          data-tooltip={intl.formatMessage({ id: 'mensClothing' })}
          data-tooltip-direction="right"
          lang={locale}
        >
          <MensClothingIcon />
          <span className={styles.linkText}>
            <FormattedMessage id="mensClothing" />
          </span>
        </Link>
        <Link
          to="/shop/women"
          data-test-id="womenCategory"
          className={styles.navLink}
          data-target="women"
          data-tooltip={intl.formatMessage({ id: 'womenClothing' })}
          data-tooltip-direction="right"
        >
          <WomenClothingIcon />
          <span className={styles.linkText} lang={locale}>
            <FormattedMessage id="womenClothing" />
          </span>
        </Link>
        <Link
          to="/shop/jewellery"
          data-test-id="jewelleryCategory"
          className={styles.navLink}
          data-target="jewellery"
          data-tooltip={intl.formatMessage({ id: 'jewellery' })}
          data-tooltip-direction="right"
          lang={locale}
        >
          <JewelleryIcon />
          <span className={styles.linkText}>
            <FormattedMessage id="jewellery" />
          </span>
        </Link>
        <Link
          to="/shop/electronics"
          data-test-id="electronicsCategory"
          className={styles.navLink}
          data-target="electronics"
          data-tooltip={intl.formatMessage({ id: 'electronics' })}
          data-tooltip-direction="right"
          lang={locale}
        >
          <ElectronicsIcon />
          <span className={styles.linkText}>
            <FormattedMessage id="electronics" />
          </span>
        </Link>
      </div>
      <Link
        to="/referrals"
        data-test-id="referralsPage"
        className={styles.referralCard}
        lang={locale}
        data-tooltip={intl.formatMessage({ id: 'referral_message' })}
        data-tooltip-direction="right"
      >
        <p>
          <FormattedMessage id="referral_message" />
        </p>
      </Link>
      <div className={styles.bottomLinks}>
        {isLoggedIn ?
          <Link 
            to="/user/profile" 
            className={`${styles.loggedInUserWrap} ${styles.bottomLink}`}
            aria-label={`${currentUser.name}'s profile`} 
          >
            <div className={styles.profileImage} style={{ backgroundImage: `url('${currentUser.photo}')` }}></div>
            <div className={styles.displayName}>
              {currentUser.name}
            </div> 
          </Link>
          :
          <button
            aria-label="Login"
            className={`${styles.loginButton} ${styles.bottomLink}`}
            data-tooltip={intl.formatMessage({ id: 'login' })}
            data-tooltip-direction="right"
            lang={locale}
            onClick={openLogin}
          >
            <LogInIcon />
            <span className={styles.linkText}>
              <FormattedMessage id="login" />
            </span>
          </button>
        }

        {isLoggedIn ? 
          <button 
            aria-label="Logout" 
            className={`${styles.logout} ${styles.bottomLink}`}
            onClick={handleLogout}
          >
            <LogoutIcon />
            <span className={styles.linkText}>
              Logout
            </span>
          </button>
          : null 
        }
        <Link
          to="/help"
          data-test-id="helpPage"
          className={`${styles.logout} ${styles.bottomLink}`}
          data-tooltip={intl.formatMessage({ id: 'help_center' })}
          data-tooltip-direction="right"
          lang={locale}
        >
          <HelpIcon />
          <span className={styles.linkText}>
            <FormattedMessage id="help_center" />
          </span>
        </Link>
      </div>
    </div>
  );
};

export default Sidebar;
