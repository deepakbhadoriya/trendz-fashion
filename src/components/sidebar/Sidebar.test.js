import React from 'react';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

import Sidebar from './Sidebar';
import TestEnvironmentWrapper from '../../test/TestEnvironmentWrapper';

const { mount } = Enzyme;
Enzyme.configure({ adapter: new Adapter() });

describe('<Sidebar/>', () => {
  it('render correctly (snapshot)', () => {
    const tree = mount(
      <TestEnvironmentWrapper>
        <Sidebar />
      </TestEnvironmentWrapper>
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
