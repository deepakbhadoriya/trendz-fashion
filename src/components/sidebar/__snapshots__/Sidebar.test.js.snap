// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar/> render correctly (snapshot) 1`] = `
<TestEnvironmentWrapper>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <IntlProvider
          defaultFormats={Object {}}
          defaultLocale="en"
          formats={Object {}}
          locale="en"
          messages={
            Object {
              "electronics": "Electronics",
              "fifty_percent_off": "Get up to 50%",
              "help_center": "Help Center",
              "holiday_season": "For the holiday season",
              "jewellery": "Jewellery",
              "login": "Log in",
              "mensClothing": "Men's clothing",
              "referral_message": "Invite a friend and get a discount of 5%",
              "register": "Register",
              "search": "Search for products",
              "thirty_percent_discount_shirts": "Get up to 30% off shirts",
              "womenClothing": "Women's Clothing",
            }
          }
          onError={[Function]}
          textComponent={Symbol(react.fragment)}
        >
          <Sidebar>
            <div
              className="sidebar sidebarFull"
            >
              <button
                aria-label="Sidebar toggle"
                className="sidebarToggler "
                onClick={[Function]}
              >
                <ArrowLeftIcon>
                  <svg
                    fill="none"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.5 19L8.5 12L15.5 5"
                      stroke="#200E32"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="1.5"
                    />
                  </svg>
                </ArrowLeftIcon>
              </button>
              <div
                className="navLinks"
              >
                <Link
                  className="navLink"
                  data-target="mens"
                  to="/men"
                >
                  <LinkAnchor
                    className="navLink"
                    data-target="mens"
                    href="/men"
                    navigate={[Function]}
                  >
                    <a
                      className="navLink"
                      data-target="mens"
                      href="/men"
                      onClick={[Function]}
                    >
                      <MensClothingIcon>
                        <svg
                          fill="none"
                          viewBox="0 0 187 187"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <circle
                            cx="93.0627"
                            cy="93.0627"
                            fill="url(#MensClothingIconGradientFilter)"
                            r="93.0627"
                          />
                          <path
                            clipRule="evenodd"
                            d="M117.139 42.0894L152.883 71.4691C153.262 71.7813 133.858 86.6057 131.51 85.5716C129.934 84.8773 126.782 83.4893 125.207 82.795C121.24 81.0481 127.427 137.267 124.458 143.739C124.275 144.136 61.8514 144.136 61.6677 143.739C58.6994 137.267 64.8861 81.0481 60.9193 82.795C59.3433 83.4893 56.1919 84.8773 54.6159 85.5716C52.2675 86.6057 32.8631 71.7813 33.2422 71.4691L68.9862 42.0894H84.3132C79.1385 56.0868 108.66 56.0868 103.485 42.0894H117.139Z"
                            fill="url(#MensClothingIconClothFilter)"
                            fillRule="evenodd"
                          />
                          <defs>
                            <linearGradient
                              gradientUnits="userSpaceOnUse"
                              id="MensClothingIconGradientFilter"
                              x1="93.0627"
                              x2="93.0627"
                              y1="0"
                              y2="186.125"
                            >
                              <stop
                                stopColor="#FFF48C"
                              />
                              <stop
                                offset="1"
                                stopColor="#FFE600"
                                stopOpacity="0.25"
                              />
                            </linearGradient>
                            <linearGradient
                              gradientUnits="userSpaceOnUse"
                              id="MensClothingIconClothFilter"
                              x1="93.0628"
                              x2="93.0628"
                              y1="42.0894"
                              y2="144.036"
                            >
                              <stop
                                stopColor="#FF8181"
                              />
                              <stop
                                offset="1"
                                stopColor="#FFF59B"
                                stopOpacity="0.62"
                              />
                            </linearGradient>
                          </defs>
                        </svg>
                      </MensClothingIcon>
                      <span
                        className="linkText"
                      >
                        <FormattedMessage
                          id="mensClothing"
                        >
                          Men's clothing
                        </FormattedMessage>
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  className="navLink"
                  data-target="women"
                  to="/women"
                >
                  <LinkAnchor
                    className="navLink"
                    data-target="women"
                    href="/women"
                    navigate={[Function]}
                  >
                    <a
                      className="navLink"
                      data-target="women"
                      href="/women"
                      onClick={[Function]}
                    >
                      <WomenClothingIcon>
                        <svg
                          fill="none"
                          viewBox="0 0 187 187"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <circle
                            cx="93.5"
                            cy="93.0627"
                            fill="url(#WomenClothingIconGradientFilter)"
                            r="93.0627"
                          />
                          <path
                            d="M131.168 71.7828C131.168 71.7828 120.515 66.5372 120.515 39.884H109.863C109.863 39.884 106.369 61.1498 93.8693 61.1498C81.37 61.1498 77.8759 39.884 77.8759 39.884H67.2231C67.2231 66.0835 56.5703 71.7828 56.5703 71.7828L72.5637 98.3792L56.5703 146.241H131.14L115.175 98.3792L131.168 71.7828Z"
                            fill="url(#WomenClothingIconClothGradientFilter)"
                          />
                          <defs>
                            <linearGradient
                              gradientUnits="userSpaceOnUse"
                              id="WomenClothingIconGradientFilter"
                              x1="93.5"
                              x2="93.5"
                              y1="0"
                              y2="186.125"
                            >
                              <stop
                                stopColor="#FF00CF"
                                stopOpacity="0.13"
                              />
                              <stop
                                offset="1"
                                stopColor="#FF0B8F"
                                stopOpacity="0.63"
                              />
                            </linearGradient>
                            <linearGradient
                              gradientUnits="userSpaceOnUse"
                              id="WomenClothingIconClothGradientFilter"
                              x1="93.8693"
                              x2="93.5"
                              y1="39.884"
                              y2="146.241"
                            >
                              <stop
                                stopColor="#FF5C00"
                              />
                              <stop
                                offset="1"
                                stopColor="#FF80C4"
                                stopOpacity="0.35"
                              />
                            </linearGradient>
                          </defs>
                        </svg>
                      </WomenClothingIcon>
                      <span
                        className="linkText"
                      >
                        <FormattedMessage
                          id="womenClothing"
                        >
                          Women's Clothing
                        </FormattedMessage>
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  className="navLink"
                  data-target="jewellery"
                  to="/jewellery"
                >
                  <LinkAnchor
                    className="navLink"
                    data-target="jewellery"
                    href="/jewellery"
                    navigate={[Function]}
                  >
                    <a
                      className="navLink"
                      data-target="jewellery"
                      href="/jewellery"
                      onClick={[Function]}
                    >
                      <JewelleryIcon>
                        <svg
                          fill="none"
                          viewBox="0 0 187 187"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <circle
                            cx="93.9372"
                            cy="93.0627"
                            fill="url(#JewelleryIconGradientFilter)"
                            r="93.0627"
                          />
                          <path
                            d="M140.254 78.885C136.465 77.5085 132.895 75.6135 129.665 73.2197C129.525 73.12 129.405 73.0202 129.266 72.9006C124.659 84.5103 115.046 93.3672 103.26 96.5589C103.26 96.519 103.26 96.4791 103.26 96.4392C103.26 94.4045 102.921 92.8286 102.323 91.6118C113.67 88.9986 122.944 81.4383 127.451 71.4643C127.152 71.205 126.833 70.9457 126.534 70.6863C126.494 70.6464 126.454 70.6065 126.414 70.5866C120.012 64.8615 116.462 56.7427 116.462 48.3246C116.462 45.7114 114.269 43.577 111.556 43.577H75.5795C72.8672 43.577 70.6735 45.7114 70.6735 48.3246C70.6735 56.7626 67.1237 64.8815 60.702 70.6065C60.6622 70.6464 60.6223 70.6863 60.5824 70.7063C60.2832 70.9656 59.9841 71.2249 59.685 71.4643C64.1921 81.4383 73.4854 88.9986 84.813 91.6118C84.2147 92.8286 83.8757 94.4045 83.8757 96.4392C83.8757 96.4791 83.8757 96.519 83.8757 96.5788C72.0894 93.3871 62.477 84.5302 57.8701 72.9205C57.7305 73.0202 57.6109 73.1399 57.4713 73.2397C54.2405 75.6334 50.6708 77.5285 46.8816 78.9049C44.7677 79.6629 43.4714 81.7375 43.7506 83.8919C45.2263 95.6213 52.5653 130.431 87.5253 142.3C91.4341 143.616 95.7018 143.616 99.6107 142.3C134.571 130.431 141.91 95.6213 143.385 83.8919C143.665 81.7375 142.368 79.6629 140.254 78.885ZM93.568 90.6343C97.9753 90.6343 100.069 91.512 100.069 96.4392C100.069 101.925 97.0978 106.573 93.568 106.573C90.0381 106.573 87.0666 101.945 87.0666 96.4392C87.0666 91.512 89.1606 90.6343 93.568 90.6343Z"
                            fill="url(#JewelleryIconIconFilter)"
                          />
                          <defs>
                            <linearGradient
                              gradientUnits="userSpaceOnUse"
                              id="JewelleryIconGradientFilter"
                              x1="93.9372"
                              x2="93.9372"
                              y1="0"
                              y2="186.125"
                            >
                              <stop
                                stopColor="#FAFF00"
                                stopOpacity="0.89"
                              />
                              <stop
                                offset="1"
                                stopColor="#63E100"
                                stopOpacity="0.3"
                              />
                            </linearGradient>
                            <linearGradient
                              gradientUnits="userSpaceOnUse"
                              id="JewelleryIconIconFilter"
                              x1="93.568"
                              x2="93.568"
                              y1="43.577"
                              y2="143.287"
                            >
                              <stop
                                stopColor="#FAC300"
                              />
                              <stop
                                offset="1"
                                stopColor="#FF3131"
                                stopOpacity="0.67"
                              />
                            </linearGradient>
                          </defs>
                        </svg>
                      </JewelleryIcon>
                      <span
                        className="linkText"
                      >
                        <FormattedMessage
                          id="jewellery"
                        >
                          Jewellery
                        </FormattedMessage>
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  className="navLink"
                  data-target="electronics"
                  to="/electronics"
                >
                  <LinkAnchor
                    className="navLink"
                    data-target="electronics"
                    href="/electronics"
                    navigate={[Function]}
                  >
                    <a
                      className="navLink"
                      data-target="electronics"
                      href="/electronics"
                      onClick={[Function]}
                    >
                      <ElectronicsIcon>
                        <svg
                          fill="none"
                          viewBox="0 0 187 187"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <circle
                            cx="93.0627"
                            cy="93.0627"
                            fill="url(#ElectronicsIconGradientFilter)"
                            r="93.0627"
                          />
                          <path
                            d="M131.601 84.147C133.252 80.3621 133.939 78.8294 133.532 73.156C132.871 63.9353 128.986 55.8435 124.203 50.3902C117.926 43.2333 106.181 39.8836 94.3872 40.0111L99.2135 70.4323C107.547 72.4107 113.755 79.8812 113.755 88.82C113.755 99.2669 105.287 107.735 94.84 107.735C84.3942 107.735 75.9246 99.2669 75.9246 88.82C75.9246 82.1487 79.3866 76.2963 84.6037 72.9291L79.7049 42.0481C74.3112 43.5609 69.4709 45.8939 65.8626 49.0248C59.0052 54.9731 51.7852 64.2898 51.5383 76.136C51.2259 91.1085 57.9218 100.777 64.0713 110.156C66.0405 113.156 68.7092 116.785 69.1866 120.387C70.1811 127.905 69.0918 134.045 65.9398 138.802C63.7309 142.137 61.7302 145.038 59.6218 147.652H109.588C107.126 143.114 106.257 135.633 109.679 135.641C114.748 135.787 118.361 137.01 123.415 136.245C127.231 135.666 129.588 134.615 130.321 131.13C131.179 127.047 128.916 126.502 129.298 124.478C129.822 121.685 131.541 120.299 132.411 117.06C133.594 112.658 131.419 109.995 133.284 109.094C135.746 107.549 137.891 107.645 140.038 105.55C143.592 102.222 129.672 89.0879 131.601 84.147Z"
                            fill="url(#ElectronicsIconIconFilter)"
                          />
                          <defs>
                            <linearGradient
                              gradientUnits="userSpaceOnUse"
                              id="ElectronicsIconGradientFilter"
                              x1="93.0627"
                              x2="93.0627"
                              y1="0"
                              y2="186.125"
                            >
                              <stop
                                stopColor="#63B4FF"
                                stopOpacity="0.89"
                              />
                              <stop
                                offset="1"
                                stopColor="#D10000"
                                stopOpacity="0.03"
                              />
                            </linearGradient>
                            <linearGradient
                              gradientUnits="userSpaceOnUse"
                              id="ElectronicsIconIconFilter"
                              x1="96.0747"
                              x2="96.0747"
                              y1="40.0076"
                              y2="147.652"
                            >
                              <stop
                                stopColor="white"
                              />
                              <stop
                                offset="1"
                                stopColor="white"
                                stopOpacity="0.46"
                              />
                            </linearGradient>
                          </defs>
                        </svg>
                      </ElectronicsIcon>
                      <span
                        className="linkText"
                      >
                        <FormattedMessage
                          id="electronics"
                        >
                          Electronics
                        </FormattedMessage>
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
              <Link
                className="referralCard"
                to="/referrals"
              >
                <LinkAnchor
                  className="referralCard"
                  href="/referrals"
                  navigate={[Function]}
                >
                  <a
                    className="referralCard"
                    href="/referrals"
                    onClick={[Function]}
                  >
                    <p>
                      <FormattedMessage
                        id="referral_message"
                      >
                        Invite a friend and get a discount of 5%
                      </FormattedMessage>
                    </p>
                  </a>
                </LinkAnchor>
              </Link>
              <div
                className="bottomLinks"
              >
                <button
                  aria-label="Logout"
                  className="logout bottomLink"
                >
                  <LogInIcon>
                    <svg
                      fill="none"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M15.8125 12.0218H3.77148"
                        stroke="#200E32"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth="1.5"
                      />
                      <path
                        d="M12.8848 9.1058L15.8128 12.0218L12.8848 14.9378"
                        stroke="#200E32"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth="1.5"
                      />
                      <path
                        d="M8.50439 7.389V6.456C8.50439 4.421 10.1534 2.772 12.1894 2.772H17.0734C19.1034 2.772 20.7484 4.417 20.7484 6.447V17.587C20.7484 19.622 19.0984 21.272 17.0634 21.272H12.1784C10.1494 21.272 8.50439 19.626 8.50439 17.597V16.655"
                        stroke="#200E32"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth="1.5"
                      />
                    </svg>
                  </LogInIcon>
                  <span
                    className="linkText"
                  >
                    <FormattedMessage
                      id="login"
                    >
                      Log in
                    </FormattedMessage>
                  </span>
                </button>
                <Link
                  className="logout bottomLink"
                  to="/help"
                >
                  <LinkAnchor
                    className="logout bottomLink"
                    href="/help"
                    navigate={[Function]}
                  >
                    <a
                      className="logout bottomLink"
                      href="/help"
                      onClick={[Function]}
                    >
                      <HelpIcon>
                        <svg
                          fill="#200E32"
                          version="1.0"
                          viewBox="0 0 5090 5090"
                          x="0px"
                          xmlns="http://www.w3.org/2000/svg"
                          y="0px"
                        >
                          <path
                            d="M2072 2085c0,110 -89,199 -198,199 -110,0 -199,-89 -199,-199 0,-240 97,-457 254,-614 157,-157 374,-254 614,-254 240,0 457,97 614,254 157,157 254,374 254,614 0,240 -97,457 -254,614 -112,112 -255,194 -415,231l0 86c0,110 -89,199 -199,199 -110,0 -199,-89 -199,-199l0 -262c0,-109 89,-198 199,-198 130,0 248,-53 333,-138 85,-85 138,-203 138,-333 0,-130 -53,-248 -138,-333 -85,-85 -203,-138 -333,-138 -130,0 -248,53 -333,138 -85,85 -138,203 -138,333zm451 1787c-110,0 -199,-88 -199,-198 0,-110 89,-199 199,-199l44 0c110,0 199,89 199,199 0,110 -89,198 -199,198l-44 0zm22 820c593,0 1130,-240 1518,-629 389,-388 629,-925 629,-1518 0,-593 -240,-1130 -629,-1519 -388,-388 -925,-629 -1518,-629 -593,0 -1130,241 -1519,629 -388,389 -629,926 -629,1519 0,593 241,1130 629,1518 389,389 926,629 1519,629zm1799 -348c-460,461 -1097,746 -1799,746 -703,0 -1339,-285 -1800,-746 -460,-460 -745,-1097 -745,-1799 0,-703 285,-1339 745,-1800 461,-460 1097,-745 1800,-745 702,0 1339,285 1799,745 461,461 746,1097 746,1800 0,702 -285,1339 -746,1799z"
                          />
                        </svg>
                      </HelpIcon>
                      <span
                        className="linkText"
                      >
                        <FormattedMessage
                          id="help_center"
                        >
                          Help Center
                        </FormattedMessage>
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </div>
          </Sidebar>
        </IntlProvider>
      </Provider>
    </Router>
  </BrowserRouter>
</TestEnvironmentWrapper>
`;
